tpr/kind_test.go

```diff
@@ -0,0 +1,43 @@
+package tpr
+
+import (
+	"testing"
+
+	"github.com/juju/errgo"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestExtractKindAndGroup(t *testing.T) {
+	tests := []struct {
+		Input         string
+		ExpectedKind  string
+		ExpectedGroup string
+		ExpectedError error
+	}{
+		{
+			Input:         "foo.company.com",
+			ExpectedKind:  "Foo",
+			ExpectedGroup: "company.com",
+		},
+		{
+			Input:         "cron-tab.company.com",
+			ExpectedKind:  "CronTab",
+			ExpectedGroup: "company.com",
+		},
+		{
+			Input:         "foo",
+			ExpectedError: unexpectedlyShortResourceNameError,
+		},
+		{
+			Input:         "foo.company",
+			ExpectedError: unexpectedlyShortResourceNameError,
+		},
+	}
+
+	for i, tt := range tests {
```

Comment:
Elsewhere I've seen this variable be called `tc`, for test case. It would be nice to standardize here, whatever we end up choosing. I think go tests introduce enough boilerplate already, and standardizing reduces the overhead of separating that from the actual logic, IMO.